tools/distpartitioning/parmetis_wrapper.py

```diff
@@ -0,0 +1,156 @@
+import argparse
+import logging
+import os
+import platform
+import sys
+from pathlib import Path
+
+import constants
+from utils import read_json
+
+
+def check_dependencies():
+    """Check if all the dependencies needed for the execution of this file
+    are installed.
+    """
+
+    exec_path = os.get_exec_path()
+    mpi_install = False
+    for x in exec_path:
+        if os.path.isfile(os.path.join(x, "mpirun")):
+            mpi_install = True
+            break
+    assert (
+        mpi_install
+    ), "Could not locate the following dependency: MPI. Please install it and try again."
+
+
+def run_parmetis_wrapper(params):
+    """Function to execute all the steps needed to run ParMETIS
+
+    Parameters:
+    -----------
+    params : argparser object
+        an instance of argparser class to capture command-line arguments
+    """
+    assert os.path.isfile(params.schema_file)
+    assert os.path.isfile(params.hostfile)
+
+    schema = read_json(params.schema_file)
+    graph_name = schema[constants.STR_GRAPH_NAME]
+    num_partitions = len(schema[constants.STR_NUM_NODES_PER_CHUNK][0])
+
+    # Check if parmetis_preprocess.py exists.
+    assert os.path.isfile(
+        os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "parmetis_preprocess.py"
+        )
+    ), "Please check DGL Installation, parmetis_preprocess.py file does not exist."
+
+    # Trigger pre-processing step to generate input files for ParMETIS.
+    preproc_cmd = (
+        f"mpirun -np {num_partitions} -hostfile {params.hostfile} "
+        f"python3 tools/distpartitioning/parmetis_preprocess.py "
+        f"--schema_file {params.schema_file} "
+        f"--output_dir {params.preproc_output_dir}"
+    )
+    logging.info(f"Executing Preprocessing Step: {preproc_cmd}")
+    os.system(preproc_cmd)
+    logging.info(f"Done Preprocessing Step")
+
+    # Trigger ParMETIS for creating metis partitions for the input graph.
+    parmetis_install_path = "pm_dglpart3"
```

Comment:
Could please share location of this `pm_dglpart3` binary?
I see only first and second version in https://github.com/KarypisLab/ParMETIS/tree/dgl/programs

Reply:
@itaraban , please use the following repository for pm_dglpart3.c. 
https://github.com/kylasa/ParMETIS.git